Code refresh + SPM support

Climbatize opened the pull request:
This PR is used in production since the last 9 month with no issues so far. (through this SPM dependency: https://github.com/Climbatize/XLPagerTabStrip.git > 9.1.0)

- Updated source code to remove warnings and use clang instead of GCC. 
- Upgraded support to iOS 11 and added support for SPM. 
- Moved FXPageControl to a SPM dependency
- Fixed deprecated code concerning window orientation.

I know there is already this good PR https://github.com/xmartlabs/XLPagerTabStrip/pull/846 aiming to bring support to SPM, but I think this lib needed a deeper refresh to remove some deprecations.

Climbatize committed:
SPM first attempt

Climbatize committed:
update FXPageControl

Climbatize committed:
added platform to SPM

Climbatize committed:
fix SPM

Climbatize committed:
attempt to fix crash

Climbatize committed:
SPM bundle module

Climbatize committed:
removed inherit module from target

Climbatize committed:
iOS < 13 support

Climbatize committed:
upgrade Example project to XCode 11

Climbatize committed:
Upgrade to official FXPageControl package

Commit:
Merge branch 'master' into master

Climbatize committed:
Readme update

mats-claassen commented on Example/Example/Helpers/PostCell.swift:
```diff
@@ -26,9 +26,9 @@ import UIKit
-    @IBOutlet weak var userImage: UIImageView!
-    @IBOutlet weak var postName: UILabel!
-    @IBOutlet weak var postText: UILabel!
+    @IBOutlet var userImage: UIImageView!
+    @IBOutlet var postName: UILabel!
+    @IBOutlet var postText: UILabel!
```
Regarding making the IBOutlets strong, could you point me to documentation stating that weak is not recommended? I read the opposite in this [doc](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/LoadingResources/CocoaNibs/CocoaNibs.html#//apple_ref/doc/uid/10000051i-CH4-SW6).
Also connecting an Outlet from Xcode defines it as weak by default

Climbatize replied:
Apple documentation is confuse, I admit. So I remade the outlets from XCode itself to be sure

https://github.com/xmartlabs/XLPagerTabStrip/assets/1507391/9824393a-62fe-400e-ae30-943f87cd1a80

Climbatize replied:
I admit Apple documentation is confusing. So I remade the IBOutlets from XCode just to confirm.


https://github.com/xmartlabs/XLPagerTabStrip/assets/1507391/8b55e26c-e020-458f-9810-fc68a3de9816

mats-claassen replied:
I have seen both happen by default. For example when you build a UIViewController and want to connect its `view` to the controller class it defaults to strong (as that `view` should be strong). But for other views which are subviews of the main `view` it will default to weak (because subviews will be strongly referenced by their parent). This means as long as a view is in the view hierarchy it should not be deallocated and you should have no issue. However if you plan to remove the view you will have to strongly reference it if you plan to continue using it afterwards.

I think it is best to revert unless there are reproducible crashes which I would rather think could be linked to how the views are used. It might make sense to switch some of the outlets to strong if the use cases require it (rather than just switching everything)

mats-claassen commented on Sources/XLPagerTabStrip/TwitterPagerTabStripViewController.swift:
```diff
@@ -23,6 +23,35 @@
+import FXPageControl
+#endif
+
+extension UIWindow {
```
I think this extension should be moved to a separate file named something like UIWindow+Pager.swift or so

mats-claassen reviewed (commented):
Hi @Climbatize thank you for contributing and for the heads up. Sorry for the delay. I left just two comments on the code changes. Did you also test installing these changes through Cocoapods?

mats-claassen commented on ObjC/FXPageControl.h:
```diff
@@ -1,7 +1,7 @@
 //  FXPageControl.h
 //
-//  Version 1.4
+//  Version 1.5
```
Moving this file out of the Sources folder will make it not be included in the library when installed through Cocoapods and it is also not marked as dependency. Please either revert or add FXPageControl as dependency in the podspec.

Climbatize replied:
That's a big mistake from my side! I ran a `pod spec lint XLPagerTabStrip.podspec` instead of `pod lib lint XLPagerTabStrip.podspec`. Fix is on the way

Climbatize committed:
PR feedbacks

Moved UIWindow extension to an independant file, updated podspec to take objective C files refactoring in account

Climbatize committed:
Upgrade FXPageControl to 1.6

Climbatize commented:
Thanks @mats-claassen for the quick and king review! I left my comments and provided update based on your feedbacks.

mats-claassen commented:
@Climbatize changes look good. Just the matter of the weak outlets remaining

Climbatize committed:
reverting weak drop

Climbatize commented:
I reverted changes on weak outlets. At first sight no crashes from my side. I think we're good to merge @mats-claassen :)

Climbatize commented:
Thanks for the merge @mats-claassen ! Don't forget to tag the master branch to a minor version greater than 9.0.0 so the release is accessible, or to edit the the readme file to tell devs to point to master branch
